### DISCUSSION TITLE
Triggers and functions

### DISCUSSION THREAD

**Discussion by immerhart** [dragonflydb/dragonfly#3153] [Q&A]
Redis deprecated triggers and functions (no idea why) and I was wondering if there are any plans to integrate something comparable into dragonfly. If the DB is used as main DB and not as cache only, its very useful to "compress" more complex queries to just one DB call... The concept of loading the function once and be able to call it later on, seems like a good approach and in general it's handy to be able to script in JS as the common usage of the language (not a JS developer :D).

**Comment by romange** [comment]
Where did you read that Redis deprecated triggers and functions?

**Reply by immerhart** [reply]
https://redis.io/docs/latest/operate/oss_and_stack/stack-with-enterprise/deprecated-features/triggers-and-functions/
There is a yellow box everywhere in the doc...

**Reply by romange** [reply]
I did not know these were deprecated in any case there is lua API that helps exactly with the use-case of `it's very useful to "compress" more complex queries to just one DB call.` Check out EVAL and EVALSHA calls.
